pkg/kubelet/dockertools/exec.go

```diff
@@ -110,6 +115,14 @@ func (*NativeExecHandler) ExecInContainer(client DockerInterface, container *doc
 	if err != nil {
 		return fmt.Errorf("failed to exec in container - Exec setup failed - %v", err)
 	}
+	if timeout > 0 {
```

Comment:
Since `client.StartExec` blocks until the process terminates, we need to place this block between `client.CreateExec` and `client.StartExec`.